mac_debug tests are failing due to skipping tests and not being able to find results

The mac_debug tests in jenkins are failing constantly due to all of the tests being skipped and then not being able to find any results. 

One of the failures is linked here with an excerpt of the console log:
https://ci.dot.net/job/dotnet_roslyn/view/Official%20Builds/job/master/job/mac_debug/2113/
```csharp
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/BasicCodeStyleTests/netcoreapp2.0/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpCodeStyleTests/netcoreapp2.0/Microsoft.CodeAnalysis.CSharp.CodeStyle.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpCompilerSymbolTest/netcoreapp2.0/Roslyn.Compilers.CSharp.Symbol.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpCompilerSyntaxTest/netcoreapp2.0/Roslyn.Compilers.CSharp.Syntax.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CSharpScriptingTest/netcoreapp2.0/Microsoft.CodeAnalysis.CSharp.Scripting.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/CodeStyleTests/netcoreapp2.0/Microsoft.CodeAnalysis.CodeStyle.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/ScriptingTest/netcoreapp2.0/Microsoft.CodeAnalysis.Scripting.UnitTests.dll
08:00:41 Skipping /Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug/Binaries/Debug/UnitTests/VBCSCompilerTests/netcoreapp2.0/Roslyn.Compilers.CompilerServer.UnitTests.dll
08:00:41 [xUnit] [INFO] - Starting to record.
08:00:41 [xUnit] [INFO] - Processing xUnit.Net-v2 (default)
08:00:41 [xUnit] [INFO] - [xUnit.Net-v2 (default)] - No test report file(s) were found with the pattern '**/xUnitResults/*.xml' relative to '/Users/dotnet-bot/j/workspace/dotnet_roslyn/master-vs-deps/mac_debug' for the testing framework 'xUnit.Net-v2 (default)'.  Did you enter a pattern relative to the correct directory?  Did you generate the result report(s) for 'xUnit.Net-v2 (default)'?
08:00:41 
[xUnit] [ERROR] - No test reports found for the metric 'xUnit.Net' with the resolved pattern '**/xUnitResults/*.xml'. Configuration error?.```

--- a/build.sh
+++ b/build.sh
@@ -122,6 +122,13 @@ then
     build_args+=" /p:BootstrapBuildPath=${bootstrap_path}"
 fi
 
+# https://github.com/dotnet/roslyn/issues/23736
+UNAME="$(uname)"
+if [[ "$UNAME" == "Darwin" ]]
+then
+    build_args+=" /p:UseRoslynAnalyzers=false"
+fi
+
 if [[ "${build}" == true ]]
 then
     echo "Building Compilers.sln"

--- a/build/scripts/tests.sh
+++ b/build/scripts/tests.sh
@@ -45,8 +45,8 @@ do
     runtimeconfig_json="${file_name%.*}".runtimeconfig.json
 
     # If the user specifies a test on the command line, only run that one
-    # "${2:-}" => take second arg, empty string if unset (regex always matches if empty)
-    if [[ ! "${file_name}" =~ "${2:-}" ]]
+    # "${2:-}" => take second arg, empty string if unset
+    if [[ ("${2:-}" != "") && (! "${file_name}" =~ "${2:-}") ]]
     then
         echo "Skipping ${file_name}"
         continue

